Task3 7v2

allmyprogramming opened the pull request:
Adding a State-Based Router

allmyprogramming committed:
Merge pull request #20 from allmyprogramming/gh-pages

movieview scss added

allmyprogramming committed:
bootstrap installed

allmyprogramming committed:
react-bootstrap card imported

allmyprogramming committed:
Form and Buttons bootstrap imported to login view.

allmyprogramming committed:
bootstrap form and button added to signup view

allmyprogramming committed:
container added to index.jsx

allmyprogramming committed:
adding rows to main-view

allmyprogramming committed:
import columns to main-view

allmyprogramming committed:
additional column code added

allmyprogramming committed:
set movie card height

allmyprogramming committed:
red border style removed from index.jsx

allmyprogramming committed:
columns added to bookcard in mainview

allmyprogramming committed:
login changed to users, unused comments removed

allmyprogramming committed:
react router dom added to packages

allmyprogramming committed:
adding state based router

allmyprogramming committed:
removed unnecessary comments

allmyprogramming committed:
adding routes to mainview

allmyprogramming committed:
adding routes mainview

allmyprogramming committed:
encodeURIComponent added

allmyprogramming committed:
onMovieClick removed

allmyprogramming committed:
created navigation bar jsx file

allmyprogramming committed:
amended navigation bar

allmyprogramming committed:
created profile view file

allmyprogramming committed:
added profile and updated movie card props

Stanley-Okwii commented on src/components/main-view/main-view.jsx:
```diff
@@ -1,86 +1,97 @@
-import { useState, useEffect } from "react";
+import React, { useState, useEffect } from "react";
+import { BrowserRouter, Routes, Route, Navigate } from "react-router-dom";
```
```suggestion
import { BrowserRouter, Routes, Route, Navigate } from "react-router";
```

Stanley-Okwii commented on src/components/navigation-bar/navigation-bar.jsx:
```diff
@@ -0,0 +1,34 @@
+import { Navbar, Container, Nav } from "react-bootstrap";
+import { Link } from "react-router-dom";
```
```suggestion
import { Link } from "react-router";
```

Stanley-Okwii commented on src/components/movie-card/movie-card.jsx:
```diff
@@ -1,22 +1,36 @@
 import React from "react";
 import PropTypes from "prop-types";
+import { Button, Card } from "react-bootstrap";
+import { Link } from "react-router-dom";
```
```suggestion
import { Link } from "react-router";
```

Stanley-Okwii commented on src/components/main-view/main-view.jsx:
```diff
@@ -1,86 +1,97 @@
-  }
+  return (
+    <BrowserRouter>
+     <NavigationBar user={user} onLoggedOut={handleLoggedOut} />
```
This line of code results in an error because `handleLoggedOut` is NOT defined. You will need to fix this by defining it.

Stanley-Okwii commented on src/components/navigation-bar/navigation-bar.jsx:
```diff
@@ -0,0 +1,34 @@
+              <>
+                <Nav.Link as={Link} to="/">Home</Nav.Link>
+                <Nav.Link as={Link} to="/profile">Profile</Nav.Link>
+                <Button variant="link" onClick={onLoggedOut}>Log Out</Button>
```
Also for this line of code, `Button` component is NOT imported, but it's being used in this file which results in an error. Please import the Button component before using it.

Stanley-Okwii reviewed (changes requested):
Hey John,

You're off to a great start. I have left a couple of comments to help you address issues within the code. Let me know if you need clarity on anything.

allmyprogramming committed:
recommended changes nav and main files

react-router-dom to react-router.  handleLoggedOut is now defined.  Button imported in navigation-bar.

allmyprogramming committed:
updated syntax error.

allmyprogramming committed:
imported profile view to main

allmyprogramming committed:
amended react-router-dom to react router only

allmyprogramming committed:
updated favorites toggle button

allmyprogramming committed:
updated routes and syntax
